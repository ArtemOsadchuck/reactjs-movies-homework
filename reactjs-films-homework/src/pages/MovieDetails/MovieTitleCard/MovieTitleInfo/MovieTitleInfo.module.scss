$start-text-align: start;

.titleWrapper {
  .heading {
    font-size: 0.8rem;
    letter-spacing: 0.5px;
    margin-bottom: 0.5rem;
    text-align: $start-text-align;
  }

  .title {
    text-align: $start-text-align;
  }
}

.otherInfoWrapper {
  display: flex;
  flex-direction: column;
  justify-content: space-between;

  .heading {
    font-size: 0.8rem;
    letter-spacing: 0.5px;
    margin-bottom: 0.2rem;
    text-align: $start-text-align;
  }

  .infoField {
    text-align: $start-text-align;
  }
}
