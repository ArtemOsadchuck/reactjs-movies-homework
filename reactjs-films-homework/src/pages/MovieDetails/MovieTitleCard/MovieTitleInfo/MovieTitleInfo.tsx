import React from 'react';

import styles from './MovieTitleInfo.module.scss';

export interface IMovieTitleInfo {
  heading: string;
  infoField: string;
}
const MovieTitleInfo: React.FC<IMovieTitleInfo> = ({ heading, infoField }) => {
  return heading === 'Title:' ? (
    <div className={styles.titleWrapper}>
      <p className={styles.heading}>{heading}</p>
      <h2 className={styles.title}>{infoField}</h2>
    </div>
  ) : (
    <div className={styles.otherInfoWrapper}>
      <p className={styles.heading}>{heading}</p>
      <p className={styles.infoField}>{infoField}</p>
    </div>
  );
};

export default MovieTitleInfo;
