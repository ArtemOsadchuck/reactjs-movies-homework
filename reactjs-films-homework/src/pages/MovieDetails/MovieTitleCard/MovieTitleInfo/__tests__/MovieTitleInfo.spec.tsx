import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import MovieTitleInfo from '../MovieTitleInfo';

describe('MovieTitleInfo', () => {
  const mockProps = {
    heading: 'Overview:',
    overview:
      'Framed in the 1940s for the double murder of his wife and her lover, upstanding banker Andy Dufresne begins a new life at the Shawshank prison',
  };

  afterEach(() => {
    cleanup();
  });

  it('MovieTitleInfo snapshot', () => {
    const { asFragment } = render(
      <MovieTitleInfo
        heading={mockProps.heading}
        infoField={mockProps.overview}
      />
    );
    expect(asFragment).toMatchSnapshot();
  });

  it('MovieTitleInfo Must have class:', () => {
    const { container } = render(
      <MovieTitleInfo
        heading={mockProps.heading}
        infoField={mockProps.overview}
      />
    );
    expect(container.firstChild).toHaveClass('otherInfoWrapper');
    expect(container.firstChild!.childNodes[0]).toHaveClass('heading');
    expect(container.firstChild!.childNodes[1]).toHaveClass('infoField');
  });
  it('MovieTitleInfo must have text:', () => {
    render(
      <MovieTitleInfo
        heading={mockProps.heading}
        infoField={mockProps.overview}
      />
    );
    expect(screen.getByText(/Overview/i)).toBeInTheDocument();
    expect(screen.getByText(/1940s/i)).toBeInTheDocument();
  });
});
