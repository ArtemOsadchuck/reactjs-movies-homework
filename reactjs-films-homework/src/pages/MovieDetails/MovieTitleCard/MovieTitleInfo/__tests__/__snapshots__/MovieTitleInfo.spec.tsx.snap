// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieTitleInfo should renders correctly 1`] = `
<DocumentFragment>
  <div
    class="otherInfoWrapper"
  >
    <p
      class="heading"
    >
      Overview:
    </p>
    <p
      class="infoField"
    >
      Framed in the 1940s for the double murder of his wife and her lover, upstanding banker Andy Dufresne begins a new life at the Shawshank prison
    </p>
  </div>
</DocumentFragment>
`;
